scripts/cast-on-tenderly/index.js

```diff
@@ -84,27 +145,35 @@ const runSpell = async function () {
     }
 
     console.info('fetching timestamp when the spell will be castable...');
-    const eta = await spell.eta();
+    const nextCastTime = await spell.nextCastTime();
+    console.info(`nextCastTime is "${nextCastTime}"`, new Date(nextCastTime.toNumber() * 1000));
 
-    console.info(`warping the time to "${eta}"...`);
-    const currentUnixTimestamp = Math.floor(Date.now() / 1000);
-    if (currentUnixTimestamp < eta) {
-        const timestampDifference = eta - currentUnixTimestamp + 1;
-        await provider.send('evm_increaseTime', [ethers.utils.hexValue(timestampDifference)]);
-    }
+    console.info(`warping the time to "${nextCastTime}"...`);
+    await provider.send('evm_setNextBlockTimestamp', [ethers.utils.hexValue(nextCastTime)]);
 
-    console.info('casting spell on a fork...');
+    console.info('casting spell...');
     try {
         const castTx = await spell.cast(DEFAULT_TRANSACTION_PARAMETERS);
         await castTx.wait();
         console.info('successfully casted');
     } catch (error) {
         console.error('casting failed', error);
     }
+};
+
+const castOnTenderly = async function (spellAddress) {
+    const spellName = await getSpellName(spellAddress);
+    console.info(`preparing to cast spell "${spellAddress}" with name "${spellName}"...`);
+
+    const { testnetId, rpcUrlPrivate } = await createTenderlyTestnet(spellName);
+
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrlPrivate);
+
+    await makeSpellTheHat(spellAddress, provider);
+
+    await sheduleWarpAndCastSpell(spellAddress, provider);
```

Comment:
Following up the comment above, in a sense, "giving the hat to the spell" is yet another implementation detail of "casting the spell" in this case.
Maybe we could embed that call into `castSpell` and have only the highest-level functions being called from the main function.

Reply:
I would disagree, at least from the usability perspective, `publishTenderlyTestnet` can be used before as well as after the casting. It also makes sense to keep `createTenderlyTestnet` and `publishTenderlyTestnet` on the same level and don't introduce props drilling (i.e. pass `testnetId` to `castSpell` just to be used by `publishTenderlyTestnet`)

Reply:
I don't think I made myself clear. I meant that the main function would have the following structure:

```javscript
const castOnTenderly = async function(spellAddress) {
    const spellName = await getSpellName(spellAddress);
    console.info(`preparing to cast spell "${spellAddress}" with name "${spellName}"...`);

    const { testnetId, rpcUrlPrivate } = await createTenderlyTestnet(spellName);

    const provider = new ethers.providers.JsonRpcProvider(rpcUrlPrivate);

    await executeSpell(spellAddress, provider);

    await publishTenderlyTestnet(testnetId);
}
```

And then inside `executeSpell` we'd have: 

```javascript
const castSpell = async function (spellAddress, provider) {
    await giveTheHatToSpell(spellAddress, provider);

    // ... 
    // ... 
    // ... 
}
```

Again, this is merely a preference, not a blocker.